### PR TITLE
Use head commit instead of merge commit for PRs

### LINKED ISSUE #10: Wrong commit sha used for PRs
### Steps to reproduce
- Have Travis CI tests run on a PR. It runs two tests by default: in-branch and merged.
- Notice that the commit sha for the merged test is not the same as the HEAD of the branch that is to be merged. Using Github lingo,  it uses `merge_commit_sha` instead.
- Notice that as a result of the wrong commit sha being used, Travieso does not report the status back to the PR, because it's tying the status to the merge commit instead.

**Expected output:** Jobs for merge tests showing up on PR.

**Actual output:** Jobs for merge tests don't show up on PR.
### Metadata

**Travieso version:**: ae3a069f3fb2f3e7fed8839c8dfc17fc4df07e70

**Python version:**: 3.5.1

**Operating system:**: Mac

### PR DESCRIPTION
**Pull request by jarekwg**
When Travis sends webhook notifications for PRs, `job['commit']` yields the merge sha instead of the head commit of the branch for the PR. Instead, `payload['head_commit']` should be used.

I've applied this fix in my local version, and it resolves the problem, however I haven't tested sufficiently to know whether you can get away with just always using `payload['head_commit']` instead of `job['commit']`, rather than in the one cases that my patch considers. Probs can I guess.

As for tests, I haven't submitted a testcase (not sure how one would work for this), but to prove my point, you can see that Travisio hasn't reported statuses back to Github for this PR!

Closes #10

### CODE REVIEW

File: travieso/views.py [hunk 1]
```diff
@@ -10,8 +10,12 @@
 @blueprint.route('/notifications', methods=['POST'])
 @authorize_travis
 def handle_notification(payload):
+    head_commit = payload['head_commit']
     for job in payload['matrix']:
         state = get_job_github_state(job['status'], job['state'])
+        # For pull requests, the commit_sha we get back is that of the merge_commit_sha.
+        # We don't want this!
+        commit = head_commit if payload['type'] == 'pull_request' else job['commit']
```
Comment (miguelfrde) [inline travieso/views.py:18]: Could you remove the line below? Otherwise this has no effect.

**Comment by miguelfrde** [comment]
Thank you for spotting this. Left just one comment, otherwise LGTM. Just curious, in your local version testing, was that line I mentioned in the comment removed?

Regarding, Travieso not reporting. It's not an issue related to this ([other](https://github.com/wizeline/travieso/pull/1) [PRs](https://github.com/wizeline/travieso/pull/3) have the statuses attached). I don't work at Wizeline anymore, so not sure if I left some token behind attached to my account or something was modified there.

I'll try this out before merging as well.

**Comment by jarekwg** [comment]
Ah yeah, sorry about that. I made the PR through github's online editor..
In my local version the line you mentioned is removed.

Regarding other PRs having tests attached: If travis also runs in-branch tests, then those would be reporting the unmerged-PR test results to the corresponding commit on the PR. I'm guessing that's what you've got in your link.

We've got in-branch testing disabled where I work, so the only tests that run are those on merge commits.

Also, now that I think about it more, I can see a possible issue arising with this if travis were to try to report both in-branch and merge-commit tests to the PR under the same TASK name. They'd override each other, with a race condition too. Before merging this there might need to be some way to differentiate the TASK name between in-branch tests and merge-commit tests... grr.

**Comment by ryanhiebert** [comment]
IMO, it's enough of an edge-case that I think this would be worthwhile to merge regardless. It's an easy edge-case to hit, but if you're hitting it it's also likely not what you really wanted to happen, and the builds are very likely going to be reporting the same thing anyway.

**Review by ryanhiebert — APPROVED** [review]

**Review by miguelfrde — APPROVED** [review]
